src/test/java/com/dawn/jat/illuminati/post/service/PostServiceTest.java

```diff
@@ -114,45 +126,55 @@ void create() {
                 .thenReturn(postEntity);
         Mockito.when(postRepository.savePost(any(PostEntity.class)))
                 .thenReturn(postEntity);
-        assertThat(postService.create(postDto), is(postEntity));
+        Mockito.when(modelMapper.map(any(), any())).thenReturn(postDto);
+        assertThat(postService.createPost(postDto), is(postDto));
         Mockito.verify(postRepository, Mockito.times(1)).savePost(postEntity);
     }
 
     @Test
     void savePost_WithDto_SuccessfullySaveDto() {
-
-        PostEntity expectedPost = new PostEntity("how-to-apply-agile-methodology",
+        PostEntity expectedPost = new PostEntity(
+                "how-to-apply-agile-methodology",
                 "How to apply Agile methodology new",
                 "How to apply Agile methodology new",
                 "01/01/2020 new",
                 new ArrayList<>(Arrays.asList("System Design", "OOP")),
-                "Li Li new");
+                "Li Li new",
+                "");
         expectedPost.setContent("new content");
-        Mockito.when(postRepository.findById(postDto.getId()))
-                .thenReturn(Optional.of(postEntity));
+        String id = postDto.getId();
+        assertNotNull(id);
+
+        Mockito.when(postRepository.existsById(id))
+                .thenReturn(true);
+
         Mockito.when(postRepository.findById(postDto.getId()))
                 .thenReturn(Optional.of(postEntity));
 
-        postService.save(postDto);
+        Mockito.when(converter.convertPostDtoToEntity(eq(postDto), any(PostEntity.class)))
+                .thenReturn(postEntity);
 
+        Mockito.when(postRepository.savePost(postEntity)).thenReturn(postEntity);
 
+        postService.savePost(postDto);
+        assertNotNull(postDto);
         Mockito.verify(postRepository, Mockito.times(1)).findById(postDto.getId());
     }
 
     @Test
-    void savePost_WithDto_ThrowPostNotFoundException() {
-        PostEntity expectedPost = new PostEntity("how-to-apply-agile-methodology",
-                "How to apply Agile methodology new",
-                "How to apply Agile methodology new",
-                "01/01/2020 new",
-                new ArrayList<>(Arrays.asList("System Design", "OOP")),
-                "Li Li new");
-        expectedPost.setContent("new content");
-        Mockito.when(postRepository.findById(postDto.getId()))
-                .thenReturn(Optional.empty());
+    void savePost_WithDto_getIdNullThrowPostCanNotBeSavedException_invalidPost() {
```

Comment:
the title of test case name we are using are followed the convention
`[function_name]_[context]_[result]`
Sometimes, we can ignore/omit the result or context, due to may be the result and context is just too general. For these cases like this we should have an detail version following the convention.
My suggest is we rename this to a shorter version
`savePost_WithIdNullDto_ThrowPostCanNotBeSavedException`